types/backbone/index.d.ts

```diff
@@ -243,10 +253,10 @@ declare namespace Backbone {
          * any instantiation logic is run for the Model.
          * @see https://backbonejs.org/#Model-preinitialize
          */
-        preinitialize(attributes?: T, options?: any): void;
+        preinitialize(attributes?: T, options?: ModelConstructorOptions<this>): void;
 
-        constructor(attributes?: T, options?: any);
-        initialize(attributes?: T, options?: any): void;
+        constructor(attributes?: T, options?: ModelConstructorOptions);
+        initialize(attributes?: T, options?: ModelConstructorOptions<this>): void;
```

Comment:
I recognize this is modeled after `ViewOptions`, but I don't like it (I don't like `ViewOptions` either and I've actually been wanting to change that for a while). It is quite common to add extra options, and with a setup like this, that means you have to import `ModelConstructorOptions`, declare an interface that extends it, and then explicitly specify that both the constructor and the `initialize` method accept your overridden interface, even if you are not actually doing anything special in the constructor. It is also going to break a lot of code.

While I'm not opposed to constraining the options type in principle, it needs to be less cumbersome. Perhaps something like this:

```ts
type CombinedModelConstructorOptions<E, M extends Model<any, any, E>> = ModelConstructorOptions<M> & E;

class Model<T = any, S = Backbone.ModelSetOptions, E = {}> extends ModelBase implements Events {
    // ...
    constructor(attributes?: T, options?: CombinedModelConstructorOptions<E, this>);
```

The `E` here stands for "**E**xtensions to the model constructor options".

Reply:
Fixed